src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs

```diff
@@ -105,9 +108,49 @@ public async Task<IActionResult> Index(DeploymentPlanIndexOptions options, Pager
                 Pager = pagerShape
             };
 
+            model.Options.DeploymentPlansBulkAction = new List<SelectListItem>() {
+                new SelectListItem() { Text = S["Delete"], Value = nameof(DeploymentPlansBulkAction.Delete) }
+            };
+
             return View(model);
         }
 
+        [HttpPost, ActionName("Index")]
+        [FormValueRequired("submit.Filter")]
+        public ActionResult IndexFilterPOST(DeploymentPlanIndexViewModel model)
+        {
+            return RedirectToAction("Index", new RouteValueDictionary {
+                { "Options.Search", model.Options.Search }
+            });
+        }
+
+        [HttpPost, ActionName("Index")]
+        [FormValueRequired("submit.BulkAction")]
+        public async Task<ActionResult> IndexPOST(DeploymentPlanIndexOptions options, IEnumerable<int> itemIds)
+        {
+            if (itemIds?.Count() > 0)
+            {
+                var checkedItems = await _session.Query<DeploymentPlan, DeploymentPlanIndex>().Where(x => x.DocumentId.IsIn(itemIds)).ListAsync();
+                switch (options.BulkAction)
+                {
+                    case DeploymentPlansBulkAction.None:
+                        break;
+                    case DeploymentPlansBulkAction.Delete:
+                        foreach (var item in checkedItems)
+                        {
+                            _session.Delete(item);
+
+                            _notifier.Success(H["Deplyment plan {0} successfully deleted.", item.Name]);
+                        }
+                        break;
```

Comment:
I think the notification for most of the others is done once, after the loop (without the name). Not bothered either way